### ISSUE TITLE
[Core feature request] Custom runner type from string

### ISSUE THREAD

**Issue by krzema12** [krzema12/github-actions-kotlin-dsl#69]
**What feature do you need?**
Runners are already supported by the library: https://github.com/krzema12/github-actions-kotlin-dsl/blob/6e7119ed9b55a9fd096b32046fcfa26fd93ad103/library/src/main/kotlin/it/krzeminski/githubactions/domain/RunnerType.kt#L7-L25

but it's not possible to define a custom one in runtime. Use cases:
* the library lags with adding a new type
* runner is passed from e.g. strategy matrix as a string, then it's not possible to use it

**Proposed implementation**  
Instead of `enum`, make it a `sealed class`, with additional
```kotlin
class Custom(val value) : RunnerType
```

**Comment by krzema12** [comment]
Workaround (or maybe even a replacement for strategy matrix?):

```kotlin
listOf(UbuntuLatest, Windows2022).forEach { runnerType ->
    job(
        name = "build_for_$runnerType",
        runsOn = runnerType,
    ) {
    // ...
    }
}
```

which results in 
![image](https://user-images.githubusercontent.com/3110813/152485235-f1e5af5a-8c59-4977-8dc5-83ea0b67a230.png)

These individual steps are visible in YAML, while with strategy matrix they are generated in runtime, however for small collections and given that one doesn't pay too much attention to the size of the generated YAML, it works fine.

**Comment by jmfayard** [comment]
@krzema12 Is it a binary compatible change to replace the enum with a sealed class?

**Comment by krzema12** [comment]
@jmfayard I don't think we really care about binary compatibility (as far as I understand it properly). The users usually consume this library through a script, so it has to be recompiled anyway. As long as the consuming code doesn't change, and I think we could achieve it with the enum -> sealed class transition, I assume it's not a breaking change.
